R/tar-terra-wrap.R

```diff
@@ -0,0 +1,153 @@
+    drv <- get_gdal_available_driver_list("raster")
+    filetype <- rlang::arg_match0(filetype, drv$name)
+
+    check_pkg_installed("terra")
+
+    name <- targets::tar_deparse_language(substitute(name))
+
+    envir <- targets::tar_option_get("envir")
+
+    command <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(command)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    pattern <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(pattern)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    .format_terra_rast_wrap_write <- eval(substitute(function(object, path) {
+        # TODO: provide mapping of major file types to extensions
+        extension <- switch(filetype,
+                            "GTiff" = ".tif",
+                            "GPKG" = ".gpkg",
+                            "")
+        saveRDS(terra::wrapCache(object,
+                                 filename = file.path(cachedir,
+                                                      basename(path),
+                                                      paste0(basename(path),
+                                                             extension)),
+                                 filetype = filetype,
+                                 gdal = gdal,
+                                 overwrite = TRUE),
+                file = path)
+    }, list(cachedir = cachedir, filetype = filetype, gdal = gdal)))
+
+    geotargets_init_cache(name = name)
+
+    # rast_cache_init <- targets::tar_target_raw(
+    #     paste0(name, "_cache_init"),
+    #     str2expression(paste0("normalizePath(file.path(",
+    #                           shQuote(cachedir), ", ",
+    #                           shQuote(name), "))")),
+    #     format = "file"
+    # )
+
+    rast_wrap <- targets::tar_target_raw(
+        name = name,
+        command = command,
+        pattern = pattern,
+        packages = packages,
+        library = library,
+        format = targets::tar_format(
+            read = function(path) terra::unwrap(readRDS(path)),
+            write = .format_terra_rast_wrap_write,
+            marshal = function(object) terra::wrap(object),
+            unmarshal = function(object) terra::unwrap(object)
+        ),
+        repository = repository,
+        iteration = iteration,
+        error = error,
+        memory = memory,
+        garbage_collection = garbage_collection,
+        deployment = deployment,
+        priority = priority,
+        resources = resources,
+        storage = storage,
+        retrieval = retrieval,
+        cue = cue
+    )
+
+    rast_cache_files <- targets::tar_target_raw(
+        paste0(name, "_cache_files"),
+        str2expression(paste0("
+            list.files(
+                file.path(", shQuote(cachedir), ", ", shQuote(name),"),
+                full.names = TRUE,
+                recursive = TRUE
+            )")),
+        format = "file_fast",
```

Comment:
You could allow the option of either "file" or "file_fast" here, but I'm guessing it doesn't really matter since it seems like nothing will ever depend on this target.